src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -2890,16 +2934,16 @@ s32 Player_UpperAction_ChangeHeldItem(Player* this, PlayState* play) {
               (sUseHeldItem || ((this->modelAnimType != PLAYER_ANIMTYPE_3) && (play->shootingGalleryStatus == 0)))))) {
         Player_SetUpperActionFunc(this, sItemActionUpdateFuncs[this->heldItemAction]);
         this->unk_834 = 0;
-        this->unk_6AC = 0;
+        this->idleType = PLAYER_IDLE_NORMAL;
         sHeldItemButtonIsHeldDown = sUseHeldItem;
 
         return this->upperActionFunc(this, play);
     }
 
-    if (Player_CheckSpecialIdleAnim(this) != 0) {
+    if (Player_CheckFidgetAnim(this) != 0) {
```

Comment:
When `Player_CheckFidgetAnim` is checked against 0, it's checking if it's either a fidget animation or a normal idle animation right? since a normal idle animation would return -1. But the name `Player_CheckFidgetAnim` would make me think it's only checking fidget animation.

So since it distinguishes output for normal idle animation (-1), fidget animation (1+) and non-idle animation (0), should the name of this function capture all 3? Like `Player_CheckIdleAnim`?

Reply:
Its difficult because it serves two purposes. It does as you describe and says whether its a fidget animation, default idle, or neither. But if it is fidget, it returns the index (plus one) that gets used for fidget animation specific things. I find it challenging to capture both purposes in the name.

I went with the current name because of its significant on the return and how the index is used for fidget specific stuff. Im open to rename it, but dont think just `Player_CheckIdleAnim` gets that importance across.

Reply:
I guess since this can be used in any context, even when the animation is not necessarily idle, then `Player_CheckForIdleAnimation` may make sense. And it has the extra bonus the fidget animations return an index.